src/test/java/net/sf/jabref/importer/fileformat/BibteXMLImporterTest.java

```diff
@@ -0,0 +1,185 @@
+package net.sf.jabref.importer.fileformat;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.importer.OutputPrinterToNull;
+import net.sf.jabref.model.entry.BibEntry;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BibteXMLImporterTest {
+
+    @BeforeClass
+    public static void setUp() {
+        Globals.prefs = JabRefPreferences.getInstance();
+    }
+
+    @Test
+    public void testExceptionOnInputStream() throws IOException {
+        try (InputStream is = Mockito.mock(InputStream.class)) {
+            Mockito.doThrow(new IOException()).when(is).read();
+
+            BibteXMLImporter importer = new BibteXMLImporter();
+            importer.importEntries(is, new OutputPrinterToNull());
+        }
+    }
+
+    @Test
+    public void testGetFormatName() {
+        BibteXMLImporter importer = new BibteXMLImporter();
+        Assert.assertEquals("BibTeXML", importer.getFormatName());
+    }
+
+    @Test
+    public void testGetCLIId() {
+        BibteXMLImporter importer = new BibteXMLImporter();
+        Assert.assertEquals("bibtexml", importer.getCLIId());
+    }
+
+    @Test
+    public void testIsRecognizedFormatAccept() throws IOException {
+        BibteXMLImporter importer = new BibteXMLImporter();
+
+        List<String> list = Arrays.asList("BibteXMLImporter.xml", "BibteXMLImporterEmpty.xml");
+
+        for (String str : list) {
+            try (InputStream is = BibteXMLImporter.class.getResourceAsStream(str)) {
+                Assert.assertTrue(importer.isRecognizedFormat(is));
+            }
+        }
+    }
+
+    @Test
+    public void testIsRecognizedFormatReject() throws IOException {
+        BibteXMLImporter importer = new BibteXMLImporter();
+
+        List<String> list = Arrays.asList("oai2.xml", "oai22.xml", "oai23.xml");
+
+        for (String str : list) {
+            try (InputStream is = BibteXMLImporter.class.getResourceAsStream(str)) {
+                Assert.assertFalse(importer.isRecognizedFormat(is));
+            }
+        }
+    }
+
+    @Test
+    public void testImportEntries() throws IOException {
+
+        BibteXMLImporter importer = new BibteXMLImporter();
+
+        try (InputStream is = BibteXMLImporter.class.getResourceAsStream("BibteXMLImporter.xml")) {
+            List<BibEntry> entries = importer.importEntries(is, new OutputPrinterToNull());
+
+            BibEntry entry = entries.get(0);
+            Assert.assertEquals(entry.getCiteKey(), "");
+            Assert.assertEquals("article", entry.getType());
+            Assert.assertEquals("Test Article", entry.getField("article"));
```

Comment:
Normally an article does not have a field "article" (in similarly for the other types). What is saved in this field?